### PR TITLE
[C] Adding petstore sample written in C

### PR DESCRIPTION
**Pull request by PowerOfCreation**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.1.x`, `4.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Features to be implemented:

- [ ] authentication
- [x] sending data to the server(e.g. a json string)

### CODE REVIEW

**Comment by PowerOfCreation** [comment]
resolves swagger-api/swagger-codegen#8206

File: samples/client/petstore/c/api/petAPI.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	                 petIdString,
+	                 NULL);
+	pet = pet_parseFromJSON(apiClient->dataReceived);
+	if(pet == NULL) {
+		return 0;
+	} else {
+		cJSON *jsonObject = pet_convertToJSON(pet);
+		cJSON_Delete(jsonObject);
+	}
+	apiClient_free(apiClient);
```
Comment (wing328) [inline samples/client/petstore/c/api/petAPI.c]: @PowerOfCreation would it be possible to reuse the same `apiClient` across different operations/methods (e.g. `petApi_addPet`)?

We may also want to store authentication-related information (e.g HTTP username, password, API key, etc) in `apiClient`

Comment (PowerOfCreation) [reply samples/client/petstore/c/api/petAPI.c]: I have tried making it thread safe, but I could make a global variable apiClient too and reuse that in all methods. I will take a look, whether I can find a way to make it thread safe and reuse the apiClient.

Authentication related data can be stored in the apiClient when I have changed it, so it won't get created/deleted each time an operation is being performed.

**Comment by wing328** [comment]
@PowerOfCreation this should be a good starting point. I'll start the reverse engineering process this weekend.

**Pull request merged**